src/lib/pickles/common.ml

```diff
@@ -235,23 +235,64 @@ let tick_public_input_of_statement ~max_proofs_verified ~feature_flags
     (Backend.Tick.Field.Vector.length input)
     ~f:(Backend.Tick.Field.Vector.get input)
 
+let maybe_scale ~scale g x l =
+  match x with None -> l | Some fp -> scale g fp :: l
+
 let ft_comm ~add:( + ) ~scale ~endoscale ~negate
     ~verification_key:(m : _ Plonk_verification_key_evals.t) ~alpha
     ~(plonk : _ Types.Wrap.Proof_state.Deferred_values.Plonk.In_circuit.t)
-    ~t_comm =
-  let ( * ) x g = scale g x in
+    ~t_comm ~options =
+  let _ = options in
+
+  (* ignore flags for now *)
   let _, [ sigma_comm_last ] =
     Vector.split m.sigma_comm (snd (Plonk_types.Permuts_minus_1.add Nat.N1.n))
   in
+  let ( * ) x g = scale g x in
+  (* [cond_mul] conditionally scale elements based on the relevant feature
+     flags, and their values.
+
+     Whenever anything is "misaligned" between the 3 arguments, this function
+     returns [None]. *)
+  let cond_mul flag sc comm =
+    match flag with
+    | Opt.Flag.(Yes | Maybe) -> (
+        match (Opt.to_option_unsafe sc, Opt.to_option_unsafe comm) with
+        | None, _ | _, None ->
+            None
+        | Some x, Some g ->
+            Some (x * g) )
+    | Opt.Flag.No ->
+        None
+  in
+  (* [x +? xopt ] optionally adds [xopt] to [x]. This function is used in
```

Comment:
these comments are very helpful!  Ty!

Reply:
perhaps call it `maybe_add` or `cond_add` instead, for consistency.  or use `*?` for `cond_mul`

Reply:
just suggestions, feel free to take them or leave them.

Reply:
Thing is `cond_mul` has 3 parameters and that's why I haven't used an infix operator definition (I wanted to!). And the `+?`, I find, makes for a nice alignment of "additions" within the code.